### PR TITLE
ONPREM-17: Upgrade Push to support Ubuntu 22 & Node 20

### PR DESCRIPTION
**Pull request by navarro967**
This PR allows the pub-sub-server to run in ubuntu 22+ and NodeJS 20+ environments.

- Switched from uws to eiows to allow compilation on modern operation systems
- Upgraded dependencies
- Added cookie header on connection to allow sticky-session support on load balancers
- Updated deprecated function calls

### CODE REVIEW

File: lib/ConnectionManager.js [hunk 1]
```diff
@@ -58,12 +61,31 @@ class ConnectionManager {
     }
 
     _hasRoomInSocketServer(roomName){
-        return this.socketIoServer.nsps['/'].adapter.rooms[roomName];
```
Comment (yossipapi) [inline lib/ConnectionManager.js:61]: The logic change here seems to be a bit wrong 
The pre logic did not return the rooms but checked if the roomName exists in the array for rooms

File: lib/ConnectionManager.js [hunk 2]
```diff
@@ -57,13 +60,23 @@ class ConnectionManager {
...
+    _hasRoomInSocketServer(roomName) {
+        return this.socketIoServer.of(DEFAULT_NAMESPACE).adapter.rooms.get(roomName);
     }
 
      _startServer() {
         let port = KalturaConfig.config.socketio.port;
-        this.socketIoServer = require('socket.io').listen(parseInt(port, 10), { wsEngine: 'uws' });
+        const io = new Server({
+            wsEngine: require("eiows").Server,
```
Comment (yossipapi) [inline lib/ConnectionManager.js:70]: Please do the require at the top as we do with others it more clean that way.

**Comment by navarro967** [comment]
@yossipapi, tested the latest in this branch in OnPrem and QA, seems working for Slides/Polls/QnA/Live-Controller.

**Comment by amiras89** [comment]
@yossipapi the code looks legit to me
Only 2 comments I have:
1. The syntax to create the new io socket is for v3 (compared to the old v2 we had - just make sure we aware of it)
2. The added code for the "CORS headers" - nothing against it, maybe @yossipapi feel otherwise?

**Pull request merged**